app/src/lib/stores/app-store.ts

```diff
@@ -2030,9 +2032,50 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
   }
 
+  /**
+   * Push changes from latest conflicts into current multi step operation step, if needed
+   *  - i.e. - multiple instance of running in to conflicts
+   */
+  private updateMultiCommitOperationConflictsIfFound(repository: Repository) {
+    const {
+      changesState,
+      multiCommitOperationState,
+    } = this.repositoryStateCache.get(repository)
+    const { conflictState } = changesState
+
+    if (conflictState === null || multiCommitOperationState === null) {
+      return
+    }
+
+    const { step } = multiCommitOperationState
+    if (step.kind !== MultiCommitOperationStepKind.ShowConflicts) {
+      return
+    }
+
+    const { manualResolutions } = conflictState
+
+    this.repositoryStateCache.updateMultiCommitOperationState(
+      repository,
+      () => ({
+        step: { ...step, manualResolutions },
+      })
+    )
+
+    if (isRebaseConflictState(conflictState)) {
+      const { currentTip } = conflictState
+      this.repositoryStateCache.updateMultiCommitOperationState(
+        repository,
+        () => ({ currentTip })
+      )
+    }
+  }
```

Comment:
Could we merge these two in one step?

Reply:
Doesn't the `isRebaseConflictState` check apply to cherry-picking too? 🤔  Or we only rely on the `sourceBranch` for that?

Reply:
This is some of more of the stuff I would like to refactor (but haven't delved into how hard it will be.. it is conflict state touches many files). When we do a load status, we update a property on the changes state called the conflictState which currently has 3 types, Merge, CherryPick, and Rebase. In a refactor I would like it just to update the multiCommitOperationState and have generic conflict state for other app purposes. All we need from those typically is the manualResolutions for the dialog box, but in order to continue a rebase we need that currentTip. Thus for now, if it is a rebase conflict I just call it out and copy it. Cherry-pick does not need anything else from the conflict state.

Reply:
Also to note, this method has a cherry-pick and rebase specific one as well.. thus these changes "should" not impact the rebase and cherry-pick yet because I am not moving from those implementations yet.

Reply:
👍 Thanks for clarifying!